src/main/scala/cromwell/util/docker/DockerHash.scala

```diff
@@ -0,0 +1,42 @@
+package cromwell.util.docker
+
+import javax.xml.bind.DatatypeConverter
+
+import cromwell.util.StringUtil._
+
+case class DockerHash(hashType: String, hashString: String) {
+  validate()
+
+  private def validate() = {
+    DatatypeConverter.parseHexBinary(hashString)
+    val length = hashString.length
+    require(length == 8 || length == 32 || length == 64, s"unexpected hash length: $length")
+  }
+}
+
+object DockerHash {
+  /**
+    * Parses a string like "type:12345678" into a DockerHash(type, hash)
+    */
+  def apply(digest: String): DockerHash = {
+    // Reverse the digest into ("hash", "type"), or just ("hash") if no ":" is found.
+    val digestParts = digest.split(":", 2).reverse
+    val hashType = digestParts.tail.headOption.getOrElse("unknown")
+    val hashString = digestParts.head
+    DockerHash(hashType, hashString)
+  }
+
+  def fromSeq(hashCollectionType: String, dockerHashes: Seq[DockerHash]): DockerHash = {
```

Comment:
I'm having trouble understanding this method...  Also I think I agree that it should return a `Try[DockerHash]` if possible